JSIL.Libraries/Includes/Core/Types/Classes/System.Array.js

```diff
@@ -1,129 +1,400 @@
 ﻿JSIL.ImplementExternals("System.Array", function ($) {
-    $.RawMethod(true, "CheckType", JSIL.IsSystemArray);
+  $.RawMethod(true, "CheckType", JSIL.IsSystemArray);
 
-    $.RawMethod(true, "Of", function Array_Of() {
-        // Ensure System.Array is initialized.
-        var _unused = $jsilcore.System.Array.Of;
+  $.RawMethod(true, "Of", function Array_Of() {
+    // Ensure System.Array is initialized.
+    var _unused = $jsilcore.System.Array.Of;
 
-        return $jsilcore.ArrayOf.apply(null, arguments);
-    });
+    return $jsilcore.ArrayOf.apply(null, arguments);
+  });
 });
 
 JSIL.MakeClass("System.Object", "System.Array", true, [], function ($) {
-    $.SetValue("__IsArray__", true);
+  $.SetValue("__IsArray__", true);
 
-    $.RawMethod(false, "GetLength", function () {
-        return this.length;
-    });
-    $.RawMethod(false, "GetLowerBound", function () {
-        return 0;
-    });
-    $.RawMethod(false, "GetUpperBound", function () {
-        return this.length - 1;
+  $.RawMethod(false, "GetLength", function (dimension) {
+    if (this.LowerBounds || false) {
```

Comment:
Really I can use just JSIL.IsArray, as for multidimansional array we still create wrapper object (it worked the same before my PR).